pkg/reconciler/builder/builder.go

```diff
@@ -203,17 +203,28 @@ func (c *Reconciler) reconcileBuilder(ctx context.Context, builder *buildapi.Bui
 		return buildapi.BuilderRecord{}, errors.Errorf("stack %s is not ready", clusterStack.Name)
 	}
 
-	keychain, err := c.KeychainFactory.KeychainForSecretRef(ctx, registry.SecretRef{
+	builderKeychain, err := c.KeychainFactory.KeychainForSecretRef(ctx, registry.SecretRef{
 		ServiceAccount: builder.Spec.ServiceAccount(),
 		Namespace:      builder.Namespace,
 	})
 	if err != nil {
 		return buildapi.BuilderRecord{}, err
 	}
 
+	stackKeychain := builderKeychain
```

Comment:
I'm a little hesitant on falling back to the stack keychain, this feels like compromising design for backwards compatibility. That being said, I'm not too married to this idea and certainly not enough to block this PR